src/python/pants/backend/codegen/thrift/python/py_thrift_namespace_clash_check.py

```diff
@@ -96,19 +123,35 @@ def execute(self):
         '{}: [{}]'
         .format(
           namespace,
-          ', '.join('({}, {})'.format(t.address.spec, path) for (t, path) in all_paths)
-        )
-        for namespace, all_paths in clashing_namespaces.items()
-      )
-      raise self.ClashingNamespaceError("""\
+          ', '.join('({}, {})'.format(t.address.spec, path) for (t, path) in all_paths))
+        for namespace, all_paths in clashing_namespaces.items())
+      error = self.ClashingNamespaceError("""\
 Clashing namespaces for python thrift library sources detected in build graph. This will silently
 overwrite previously generated python sources with generated sources from thrift files declaring the
 same python namespace. This is an upstream WONTFIX in thrift, see:
       https://issues.apache.org/jira/browse/THRIFT-515
-
-Use --{}-skip to avoid this check if this breaks your existing build.
 Errors:
 {}
-""".format(self.get_options_scope_equivalent_flag_component(), pretty_printed_clashing))
-    else:
-      self.context.products.register_data('_py_thrift_namespaces_by_files', namespaces_by_files)
+""".format(pretty_printed_clashing))
+      if self.get_options().strict:
+        raise error
+      else:
+        self.context.log.warn(error)
+    return namespaces_by_files
+
+  def execute(self):
+    deprecated_conditional(
+      lambda: not self.get_options().strict,
+      '1.18.0.dev0',
```

Comment:
This will trigger a warning immediately for all users (in addition to the other new rendered warning).. I don't think we're quite ready for that internally.

Should probably leave a TODO to consider deprecating the flag in a few versions instead?

Reply:
Hm, ok. I might see if I can canonicalize that by extending our current deprecation methods to support a period before the deprecation warning is shown.

Reply:
Separately, I made #7494.